doc/developers-guide/Testing-MongooseIM.md

```diff
@@ -114,31 +132,78 @@ cd $MONGOOSEIM/test/ejabberd_tests/
 make quicktest TESTSPEC=my-feature.spec
 ```
 
-It's customary to create a per-feature (or per-project, if you're cloning away) `.spec` file and only enable the suites / test groups you want to test - this speeds up the iteration cycle by not testing the parts of the system that you know have not changed.
-It's worth running `default.spec` once in a while to check for regressions, though.
+Developers usually create a per-feature (or per-project, if you're cloning away) `.spec` file and only enable the suites / test groups they want to test - this speeds up the development cycle by not testing parts of the system that are not being changed.
+It's worth running `default.spec` once in a while to check for regressions.
 
-Have a look into the `default.spec` file to see how to pick only the interesting tests to run.
+Consult the `default.spec` file to see how to run only selected tests/groups/cases.
 
-If you're sure that none of the test dependencies have changed and you only edited the test suites, it's possible to speed up the tests by skipping the Rebar dependency and compilation checks by providing `PREPARE=` (i.e. an empty value):
+If you're sure that none of the test dependencies have changed and you only edited the test suites and/or MongooseIM code, it's possible to speed up the tests by skipping the Rebar dependency and compilation checks by providing `PREPARE=` (i.e. an empty value):
 
 ```sh
 make quicktest PREPARE=
 ```
 
-Have a look inside the `Makefile` to see how it works.
+Consult the `test.disabled/ejabberd_tests/Makefile` to see how it works.
 
-### Reloading node(s) code
+### Applying code changes
 
-When working on a feature or a bug fix, often you modify the code and check if it works as expected. 
-In order to change the code on dev nodes that are already generated (`mim*` and `fed*`), recompile the code for a specific node.
-For example to update the code on `mim1` node, all you have to do is:
+When working on a feature or a bug fix you often modify the code and check if it works as expected.
+In order to change the code on dev nodes that are already generated (`mim*` and `fed*`) recompile the code for a specific node.
+For example, to update the code on `mim1` node all you have to do is:
 ```sh
 ./rebar3 as mim1 compile
 ```
 
-A similar command applies to other nodes, the important thing here is rebar3's profile.
+A similar command applies to other nodes, the important thing being rebar3's profile.
+
+When the above command finishes the code can be reloaded on the server by either reloading changed module(s) in the node's shell, e.g. `l(mongoose_riak)`, or restarting the node.
+
+### Reading test reports
+
+When finished, test engine writes detailed html reports into a directory:
```

Comment:
When finished, the test engine writes the detailed html reports into a directory: